### PR TITLE
Use today as a default 'from' value

### PR DESCRIPTION
**Pull request by sschneid**

### CODE REVIEW

File: lib/vmpooler/api.rb [hunk 1]
```diff
@@ -258,8 +258,10 @@ def mean(list)
             halt 400, 'Invalid "to" date format, must match YYYY-MM-DD'
           end
 
-          # We do need a from date, however.
-          if from_param.nil? or !validate_date_str(from_param.to_s)
+          # It is also ok if from_param is nil, we will assume this also means 'today'
```
Comment (rji) [inline lib/vmpooler/api.rb]: I would assume that the `from` param always has to be some date in the past, and that the `to` param could be some date in the past _or_ 'today', if it isn't specified. We already assume that an empty `to` param is 'today' on lines 255-256.

What is the benefit of assuming that the `from` date is 'today' if the param isn't specified?

Comment (sschneid) [inline lib/vmpooler/api.rb]: It allows the param to be unset, and the default `/summary` view (with no params supplied) is a summary of the current day.

Comment (colinPL) [inline lib/vmpooler/api.rb]: Currently, a missing `from` value will result in a 40x error code. This change will allow a request to `/summary` without any parameters required and return today's information.

**Comment by colinPL** [comment]
:+1: .. awaiting response from @rji

**Comment by rji** [comment]
I think this might be cleaner if it was implemented as multiple Sinatra routes, say `/summary` and `/summary/?`, and each of those called a method that implemented this code, instead of having the code live in the route itself.

But since we don't do this anywhere else yet, I'm not going to block the PR on my comments alone. :+1:

**Pull request merged**